CMake/External_YAMLcpp.cmake

```diff
@@ -1,48 +1,43 @@
 # If we're building Boost, use that one.
 if(fletch_ENABLE_Boost)
-  set(yamlcpp_use_external_boost
-    -DYAMLCPP_USE_EXTERNAL_BOOST:BOOL=ON
-    -DBOOST_ROOT:PATH=${BOOST_ROOT}
-    )
   set(_YAMLCPP_DEPENDS ${_YAMLCPP_DEPENDS} Boost)
 else()
-  set(yamlcpp_use_external_boost
-    -DYAMLCPP_USE_EXTERNAL_BOOST:BOOL=OFF
-    )
+  message(FATAL_ERROR "YAMLcpp requires the Boost library.")
 endif()
 
 # YAMLcpp has some sort of odd build error (with clang, at least)
 # if you build the tools.
 
 ExternalProject_Add(YAMLcpp
-  DEPENDS ${_YAMPCPP_DEPENDS}
+  DEPENDS ${_YAMLCPP_DEPENDS}
   URL ${YAMLcpp_url}
   URL_MD5 ${YAMPcpp_md5}
   DOWNLOAD_NAME ${YAMLcpp_dlname}
   PREFIX  ${fletch_BUILD_PREFIX}
   DOWNLOAD_DIR ${fletch_DOWNLOAD_DIR}
   INSTALL_DIR  ${fletch_BUILD_INSTALL_PREFIX}
+  PATCH_COMMAND ${CMAKE_COMMAND}
+    -DYAMLcpp_patch:PATH=${fletch_SOURCE_DIR}/Patches/YAMLcpp
+    -DYAMLcpp_source:PATH=${fletch_BUILD_PREFIX}/src/YAMLcpp
+    -P ${fletch_SOURCE_DIR}/Patches/YAMLcpp/Patch.cmake
   CMAKE_GENERATOR ${gen}
   CMAKE_ARGS
     ${COMMON_CMAKE_ARGS}
-    ${libkml_use_external_boost}
+    -DBOOST_ROOT:PATH=${BOOST_ROOT}
     -DYAML_CPP_BUILD_CONTRIB:BOOL=OFF
     -DYAML_CPP_BUILD_TOOLS:BOOL=OFF
 )
 
 fletch_external_project_force_install(PACKAGE YAMLcpp)
 
-set(YAMLCPP_ROOT ${fletch_BUILD_INSTALL_PREFIX} CACHE STRING "" FORCE)
-set(YAMLCPP_DIR "${LIBKML_ROOT}/lib/cmake" CACHE PATH "" FORCE)
-set(YAMLCPP_LIBNAME yaml-cpp)
+set(YAMLCPP_ROOT ${fletch_BUILD_INSTALL_PREFIX})
```

Comment:
I think this variable should still go to cache in case a fletch package needs to know about yaml.

Reply:
I did not think we needed a YAMLCPP_DIR variable, do you think we need it? I think YAMLCPP_ROOT is enough... or should we have DIR and not ROOT... ?